src/Core/Utilities/ServiceCollectionExtensions.cs

```diff
@@ -502,5 +505,31 @@ public static IMvcBuilder AddViewAndDataAnnotationLocalization(this IMvcBuilder
                         return factory.Create("SharedResources", assemblyName.Name);
                     });
         }
+
+        public static IServiceCollection AddDistributedIdentityServices(this IServiceCollection services, GlobalSettings globalSettings)
+        {
+            if (globalSettings.SelfHosted || string.IsNullOrWhiteSpace(globalSettings.IdentityServer?.RedisConnectionString))
```

Comment:
I would just make this a string empty/null check on the connection string. That way self-hosted people could use distributed redis cache if they needed to for some reason.

Reply:
Done... here and in the post-configuration options, `ConfigureOpenIdConnectDistributedOptions` to the same end.